ArjunaJTA/cdi/classes/com/arjuna/ats/jta/cdi/transactional/TransactionalInterceptorBase.java

```diff
@@ -109,9 +109,11 @@ private Transactional getTransactional(InvocationContext ic) {
         // when the CDI implementation is Weld then using the Weld API for accessing the annotated type
         if (ic.getContextData().containsKey(WELD_INTERCEPTOR_BINDINGS_KEY)) {
```

Comment:
I think that "an easier" and in way of how the code is written better (feel free to correct me) fix would be like

```
// not-null for CDI && CDI is managed by Weld
if (interceptedBean != null && ic.getContextData().containsKey(WELD_INTERCEPTOR_BINDINGS_KEY)) {
```

What do you think?

Reply:
I guess checking `annotationBindings` will be an efficient one and tested it in my local not able to see any NPE.